src/geometry/geometry.cpp

```diff
@@ -39,6 +39,37 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
   auto geometry = std::make_shared<StateDescriptor>("geometry");
   Initialize<CoordSysMeshBlock>(pin, geometry.get());
 
+  Params &params = geometry->AllParams();
+
+  // Store name of geometry in parameters
+  std::string geometry_name;
+  if (typeid(PHOEBUS_GEOMETRY) == typeid(Geometry::Minkowski)) {
+    geometry_name = "Minkowski";
+  } else if (typeid(PHOEBUS_GEOMETRY) == typeid(Geometry::BoostedMinkowski)) {
+    geometry_name = "BoostedMinkowski";
+  } else if (typeid(PHOEBUS_GEOMETRY) == typeid(Geometry::SphericalMinkowski)) {
+    geometry_name = "SphericalMinkowski";
+  } else if (typeid(PHOEBUS_GEOMETRY) == typeid(Geometry::BoyerLindquist)) {
+    geometry_name = "BoyerLindquist";
+  } else if (typeid(PHOEBUS_GEOMETRY) == typeid(Geometry::SphericalKerrSchild)) {
+    geometry_name = "SphericalKerrSchild";
+  } else if (typeid(PHOEBUS_GEOMETRY) == typeid(Geometry::FLRW)) {
+    geometry_name = "FLRW";
+  } else if (typeid(PHOEBUS_GEOMETRY) == typeid(Geometry::FMKS)) {
+    geometry_name = "FMKS";
+  } else if (typeid(PHOEBUS_GEOMETRY) == typeid(Geometry::Snake)) {
+    geometry_name = "Snake";
+  } else if (typeid(PHOEBUS_GEOMETRY) == typeid(Geometry::Inchworm)) {
+    geometry_name = "Inchworm";
+  } else if (typeid(PHOEBUS_GEOMETRY) == typeid(Geometry::MonopoleSph)) {
+    geometry_name = "MonopoleSph";
+  } else if (typeid(PHOEBUS_GEOMETRY) == typeid(Geometry::MonopoleCart)) {
+    geometry_name = "MonopoleCart";
+  } else {
+    PARTHENON_THROW("PHOEBUS_GEOMETRY not recognized!");
+  }
+  params.Add("geometry_name", geometry_name);
+
```

Comment:
Can we reduce the code here by just letting the preprocessor/cmake handle it? e.g., in `compile_constants.hpp.in` add something like
```C++
#define PHOEBUS_GEOMETRY_STRING "@PHOEBUS_GEOMETRY@"
```
and then we can just do 
```C++
std::string geometry_name = PHOEBUS_GEOMETRY_STRING;
params.add("geometry_name", geometry_name);
```

Reply:
Ok I have no idea why my initial idea didn't work, but the following evil preprocessor magic *does* work:
```C++
// in a new file
#define TOKEN2STRING_IMPL(x) #x
#define TOKEN2STRING(x) TOKEN2STRING_IMPL(x)
```
and then:
```C++
// in geometry/coordinate_systems.hpp
constexpr char GEOMETRY_NAME[] = TOKEN2STRING(PHOEBUS_GEOMETRY);
```
which means `GEOMETRY_NAME` will be `Geometry::@PHOEBUS_GEOMETRY` so then we have to pull out the namespace. In `Geomery::Initialize` in `geometry/geometry.cpp` we add this:
``C++
const std::string gname = GEOMETRY_NAME;
const std::string geometry_name = gname.substr(gname.find("::") + 1);
```
and that is the correct name.